vector/src/main/java/im/vector/activity/VectorRoomCreationActivity.java

```diff
@@ -333,25 +330,24 @@ public static String isDirectChatRoomAlreadyExist(String aUserId, MXSession mSes
                 if (directChatRoomsDict.containsKey(aUserId)) {
                     ArrayList<String> roomIdsList = new ArrayList<>(directChatRoomsDict.get(aUserId));
 
-                    if (0 != roomIdsList.size()) {
+                    if (!roomIdsList.isEmpty()) {
                         for (String roomId : roomIdsList) {
                             Room room = mSession.getDataHandler().getRoom(roomId, false);
-
-                            // check if the room is already initialized
-                            if ((null != room) && room.isReady() && !room.isInvited() && !room.isLeaving()) {
-                                //dinsic: if the member is not already in matrix and just invited he's not active but
-                                // the room can be considered as ok
-                                if (!MXSession.isUserId(aUserId)){
-                                    return roomId;
-                                }
-                                else {
-                                    // test if the member did not leave the room
-                                    Collection<RoomMember> members = room.getActiveMembers();
-
-                                    for (RoomMember member : members) {
-                                        if (TextUtils.equals(member.getUserId(), aUserId)) {
-                                            Log.d(LOG_TAG, "## isDirectChatRoomAlreadyExist(): for user=" + aUserId + " roomFound=" + roomId);
-                                            return roomId;
+                            if ((null != room) && !room.isLeaving()) {
```

Comment:
You removed by mistake the `room.isReady() `check, please add it

Reply:
No it's not a mistake ; the removal of room.isReady is necessary because there is one problem of threads concurrency (callBack). That is, we never go back into the method and the pending invitation can not be accepted; which leads systematically to the creation of a new direct chat